### PR TITLE
#167377486 Setup CircleCI

### PR DESCRIPTION
**Pull request by ibkadeeko**
**What does this PR do?**
This pull request sets up circleCI for continuous Integration

**Description of Task to be completed**
- integrate circleCi with repository
- configure circleCi file

**How should this be manually tested?**
N/A

**Any Background Context You want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
[#167377486](https://pivotaltracker.com/story/show/167377486)

**Screenshots (if appropriate)** 
Screenshot 2019-08-26 at 19 48 54
Screenshot 2019-08-26 at 19 51 43

**Questions**
N/A

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["babel-preset-react-app"]
```
Comment (iverenshaguy) [inline .babelrc]: This doesn't look right. Shouldn't it be 
```suggestion
  "presets": ["react-app"]
```

Also, we won't be using CRA's preset, please change it to what Gilbert used. We're setting up from scratch, even babel.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Javascript Node CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-javascript/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version you desire here
```
Comment (iverenshaguy) [inline .circleci/config.yml]: I wonder why all these comments that do apparent things were left here, any reason?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+      - run: npm install
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+      # run tests!
```
Comment (iverenshaguy) [inline .circleci/config.yml]: ???

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    steps:
+      - checkout
+      # Download and cache dependencies
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+      - run: npm install
```
Comment (iverenshaguy) [inline .circleci/config.yml]: Please add a name and command as you did for the test step. We want to be as consistent as possible.

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+      # run tests!
+      - run:
+          name: run test
+          command: npm test
+      - run:
+          name: run tests with JUnit as reporter
+          command: npm run jest:junit
```
Comment (iverenshaguy) [inline .circleci/config.yml]: I'm confused? Who's using JUnit and how is this part of your task?

File: package.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "vali-ah-frontend",
+  "version": "1.0.0",
+  "description": "1kbIdeas - The one-stop shop for all the self-help ideas you need. =======",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest",
+    "jest:junit": "jest --ci --runInBand --reporters=default --reporters=jest-junit"
```
Comment (iverenshaguy) [inline package.json]: Please, what is the advantage of jest-JUnit reporter over the default jest reporter?

**Review by iverenshaguy — CHANGES_REQUESTED** [review]

File: .babelrc (hunk 1 above)
Comment (ibkadeeko) [reply .babelrc]: Ok

File: .circleci/config.yml (hunk 2 above)
Comment (ibkadeeko) [reply .circleci/config.yml]: Will remove them

File: .circleci/config.yml (hunk 4 above)
Comment (ibkadeeko) [reply .circleci/config.yml]: ok

File: .circleci/config.yml (hunk 5 above)
Comment (ibkadeeko) [reply .circleci/config.yml]: jUnit is how circle ci gets its own report of the test.

The first image has junit configured and the second image doesn't

Screenshot 2019-08-27 at 09 15 14
Screenshot 2019-08-27 at 09 15 41

File: package.json (hunk 6 above)
Comment (ibkadeeko) [reply package.json]: This is for the jest junit reporter. To report to circleci

File: .circleci/config.yml (hunk 5 above)
Comment (iverenshaguy) [reply .circleci/config.yml]: Really nice. So instead of just running it in the logs and sending it to code climate, we'll also see the report here?

Comment (ibkadeeko) [reply .circleci/config.yml]: Yes

File: package.json [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "vali-ah-frontend",
+  "version": "1.0.0",
+  "description": "1kbIdeas - The one-stop shop for all the self-help ideas you need. =======",
+  "main": "index.js",
+  "scripts": {
+    "test": "jest",
```
Comment (ibkadeeko) [inline package.json]: remove test

**Review by iverenshaguy — APPROVED** [review]

**Pull request merged**